### ISSUE TITLE
cell-census-builder error when building docker image on M1 Mac

### ISSUE THREAD

**Issue by bkmartinjr** [chanzuckerberg/cellxgene-census#288]
Building the container fails on an M1 Mac. Note that this is an unsupported platform for the image, but it would be useful if this worked for development purposes, once tiledbsoma wheels are available for M1 Mac).

```
=> ERROR [6/7] RUN python3 -m pip install /tools/cell_census_builder/*.whl 15.4s
------
> [6/7] RUN python3 -m pip install /tools/cell_census_builder/*.whl:
#0 0.382 Processing /tools/cell_census_builder/cell_census_builder-0.10.1.dev16+ga7fe826-py3-none-any.whl
#0 0.521 Collecting typing-extensions
#0 0.653 Downloading typing_extensions-4.5.0-py3-none-any.whl (27 kB)
#0 0.877 Collecting numpy
#0 0.898 Downloading numpy-1.24.2-cp310-cp310-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (14.0 MB)
#0 2.188 ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 14.0/14.0 MB 10.8 MB/s eta 0:00:00
#0 2.242 Collecting gitpython
#0 2.264 Downloading GitPython-3.1.31-py3-none-any.whl (184 kB)
#0 2.281 ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 184.3/184.3 kB 11.2 MB/s eta 0:00:00
#0 2.283 Requirement already satisfied: Cython in /usr/local/lib/python3.10/dist-packages (from cell-census-builder==0.10.1.dev16+ga7fe826) (0.29.33)
#0 2.319 Collecting anndata>=0.8
#0 2.341 Downloading anndata-0.8.0-py3-none-any.whl (96 kB)
#0 2.349 ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 96.1/96.1 kB 11.9 MB/s eta 0:00:00
#0 2.383 Collecting fsspec
#0 2.405 Downloading fsspec-2023.3.0-py3-none-any.whl (145 kB)
#0 2.417 ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 145.4/145.4 kB 12.4 MB/s eta 0:00:00
#0 2.418 Requirement already satisfied: requests in /usr/lib/python3/dist-packages (from cell-census-builder==0.10.1.dev16+ga7fe826) (2.25.1)
#0 2.418 Requirement already satisfied: wheel in /usr/local/lib/python3.10/dist-packages (from cell-census-builder==0.10.1.dev16+ga7fe826) (0.40.0)
#0 2.418 Requirement already satisfied: pyyaml in /usr/lib/python3/dist-packages (from cell-census-builder==0.10.1.dev16+ga7fe826) (5.4.1)
#0 2.445 Collecting owlready2
#0 2.468 Downloading Owlready2-0.40.tar.gz (27.3 MB)
#0 4.932 ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 27.3/27.3 MB 10.1 MB/s eta 0:00:00
#0 5.322 Preparing metadata (setup.py): started
#0 5.517 Preparing metadata (setup.py): finished with status 'done'
#0 5.622 Collecting scipy
#0 5.642 Downloading scipy-1.10.1-cp310-cp310-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (30.9 MB)
#0 8.422 ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 30.9/30.9 MB 10.0 MB/s eta 0:00:00
#0 8.514 Collecting pyarrow
#0 8.542 Downloading pyarrow-11.0.0-cp310-cp310-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (32.7 MB)
#0 11.47 ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 32.7/32.7 MB 10.0 MB/s eta 0:00:00
#0 11.53 Collecting cell-census==0.10.0
#0 11.55 Downloading cell_census-0.10.0-py3-none-any.whl (11 kB)
#0 11.59 Collecting s3fs
#0 11.61 Downloading s3fs-2023.3.0-py3-none-any.whl (27 kB)
#0 11.65 Collecting attrs>=22.2.0
#0 11.67 Downloading attrs-22.2.0-py3-none-any.whl (60 kB)
#0 11.67 ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 60.0/60.0 kB 7.0 MB/s eta 0:00:00
#0 11.78 Collecting pandas
#0 11.81 Downloading pandas-1.5.3-cp310-cp310-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (11.4 MB)
#0 12.83 ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 11.4/11.4 MB 11.0 MB/s eta 0:00:00
#0 12.94 Collecting psutil
#0 12.96 Downloading psutil-5.9.4.tar.gz (485 kB)
#0 13.00 ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 485.8/485.8 kB 11.3 MB/s eta 0:00:00
#0 13.04 Installing build dependencies: started
#0 14.35 Installing build dependencies: finished with status 'done'
#0 14.35 Getting requirements to build wheel: started
#0 14.47 Getting requirements to build wheel: finished with status 'done'
#0 14.48 Preparing metadata (pyproject.toml): started
#0 14.59 Preparing metadata (pyproject.toml): finished with status 'done'
#0 14.82 Collecting aio
#0 14.84 Downloading aiohttp-3.8.4-cp310-cp310-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (1.0 MB)
#0 14.95 ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.0/1.0 MB 9.7 MB/s eta 0:00:00
#0 14.98 Collecting scikit-misc
#0 15.01 Downloading scikit-misc-0.1.4.tar.gz (852 kB)
#0 15.09 ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 852.0/852.0 kB 11.1 MB/s eta 0:00:00
#0 15.12 Preparing metadata (setup.py): started
#0 15.20 Preparing metadata (setup.py): finished with status 'error'
#0 15.20 error: subprocess-exited-with-error
#0 15.20
#0 15.20 × python setup.py egg_info did not run successfully.
#0 15.20 │ exit code: 1
#0 15.20 ╰─> [8 lines of output]
#0 15.20 Traceback (most recent call last):
#0 15.20 File "<string>", line 2, in <module>
#0 15.20 File "<pip-setuptools-caller>", line 34, in <module>
#0 15.20 File "/tmp/pip-install-rb0tfsxy/scikit-misc_622dead788df4f98bc97277a211f358f/setup.py", line 170, in <module>
#0 15.20 setup_package()
#0 15.20 File "/tmp/pip-install-rb0tfsxy/scikit-misc_622dead788df4f98bc97277a211f358f/setup.py", line 144, in setup_package
#0 15.20 from numpy.distutils.core import setup, numpy_cmdclass
#0 15.20 ModuleNotFoundError: No module named 'numpy'
#0 15.20 [end of output]
#0 15.20
#0 15.20 note: This error originates from a subprocess, and is likely not a problem with pip.
#0 15.20 error: metadata-generation-failed
#0 15.20
#0 15.20 × Encountered error while generating package metadata.
#0 15.20 ╰─> See above for output.
#0 15.20
#0 15.20 note: This is an issue with the package mentioned above, not pip.
#0 15.20 hint: See above for details.
------
ERROR: failed to solve: executor failed running [/bin/sh -c python3 -m pip install /tools/cell_census_builder/*.whl]: exit code: 1
make: *** [image] Error 1
```

**Comment by atolopko-czi** [comment]
```
❯ git sha1
a7fe82660a69cf85a1c528a9e70d044be3abcc4f

❯ pip list
Package             Version
------------------- -------------------------------
aiobotocore         2.4.2
aiohttp             3.8.4
aioitertools        0.11.0
aiosignal           1.3.1
anndata             0.8.0
async-timeout       4.0.2
attrs               22.2.0
botocore            1.27.59
build               0.10.0
cell-census         0.10.0
cell-census-builder 0.10.1.dev51+ge4f4c57.d20230324
certifi             2022.12.7
cfgv                3.3.1
charset-normalizer  2.1.1
contourpy           1.0.7
cycler              0.11.0
Cython              0.29.33
distlib             0.3.6
exceptiongroup      1.1.0
filelock            3.9.0
fonttools           4.38.0
frozenlist          1.3.3
fsspec              2023.3.0
gitdb               4.0.10
GitPython           3.1.31
h5py                3.8.0
html2text           2020.1.16
identify            2.5.17
idna                3.4
iniconfig           2.0.0
jmespath            1.0.1
joblib              1.2.0
kiwisolver          1.4.4
llvmlite            0.39.1
lxml                4.9.2
matplotlib          3.6.3
multidict           6.0.4
natsort             8.2.0
networkx            3.0
nodeenv             1.7.0
numba               0.56.4
numpy               1.23.5
Owlready2           0.40
pa                  0.0.23.dev0
packaging           23.0
pandas              1.5.3
patsy               0.5.3
Pillow              9.4.0
pip                 22.2.1
platformdirs        3.0.0
pluggy              1.0.0
pre-commit          3.0.4
psutil              5.9.4
pyarrow             11.0.0
pynndescent         0.5.8
pyparsing           3.0.9
pyproject_hooks     1.0.0
pytest              7.2.1
python-dateutil     2.8.2
pytz                2022.7.1
PyYAML              6.0
requests            2.28.2
requests-mock       1.10.0
s3fs                2023.3.0
scanpy              1.9.2
scikit-learn        1.2.1
scikit-misc         0.1.4
scipy               1.10.1
seaborn             0.12.2
session-info        1.0.0
setuptools          63.2.0
six                 1.16.0
smmap               5.0.0
somacore            1.0.0
statsmodels         0.13.5
stdlib-list         0.8.0
threadpoolctl       3.1.0
tiledb              0.20.0
tiledbsoma          1.0.0
tomli               2.0.1
tqdm                4.64.1
typing_extensions   4.4.0
umap-learn          0.5.3
UNKNOWN             0.0.0
urllib3             1.26.14
virtualenv          20.19.0
wheel               0.38.4
wrapt               1.14.1
yarl                1.8.2
```

**Comment by pablo-gar** [comment]
Support for M1/M2 macos has been resolved elsewhere #473
